src/cts/src/HTreeBuilder.cpp

```diff
@@ -123,6 +123,14 @@ void HTreeBuilder::preSinkClustering(
                                   options_->getSinkBuffer(),
                                   legalCenter.getX() * wireSegmentUnit_,
                                   legalCenter.getY() * wireSegmentUnit_);
+      if (logger_->debugCheck(CTS, "legalizer", 2) && (center != legalCenter)) {
```

Comment:
You don't need the `logger_->debugCheck(CTS, "legalizer", 2)` in the if statement. The `debugPrint` function already checks this for you. Here and in the other debugPrints.

Reply:
I want to issue debugPrint only if legalizer moved things (center != legalCenter).  Is there an alternate way without debugCheck?

Reply:
You can keep the `center != legalCenter`, but remove the `logger_->debugCheck(CTS, "legalizer", 2)` from the if statement.

Reply:
Good idea.  Done.